src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,29 @@
+package mate.academy.spring.dao.impl;
+
+import static mate.academy.spring.model.Role.RoleName;
+
+import java.util.Optional;
+import mate.academy.spring.dao.AbstractDao;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.Role;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl extends AbstractDao<Role> implements RoleDao {
+    public RoleDaoImpl(SessionFactory factory) {
+        super(factory, Role.class);
+    }
+
+    @Override
+    public Optional<Role> getByName(RoleName roleName) {
+        try (Session session = factory.openSession()) {
+            return session.createQuery("from Role where roleName = :roleName", Role.class)
+                    .setParameter("roleName", roleName).uniqueResultOptional();
```

Comment:
for better readability
```suggestion
                    .setParameter("roleName", roleName)
                    .uniqueResultOptional();
```